### PR TITLE
feat(popover): add anchor directive

### LINKED ISSUE #199: [Popover] Add PopoverAnchor directive
### Overview

https://www.radix-ui.com/primitives/docs/components/popover#anchor

### Examples in other libraries

https://originui.com/dropdowns-popovers (last example)

How use it:
https://github.com/origin-space/originui/blob/main/components/popovers/popover-09.tsx

### PR DESCRIPTION
**Pull request by pawel-twardziak**
Closes #199

### Description

This PR adds an anchor directive to `popover`

### CODE REVIEW

**Comment by pawel-twardziak** [comment]
@pimenovoleg draft. works, some minor issue related to the trigger and the anchor as the overlay origin - click from outside the content and the origin conflict

File: packages/primitives/popover/src/popover-anchor.directive.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        this.emitOutsideClick();
+    }
+
+    /** @ignore */
+    setPopoverRoot(popoverRoot: RdxPopoverRootDirective) {
+        this.popoverRoot.set(popoverRoot);
+    }
+
+    private emitOutsideClick() {
+        const clickEvent = new MouseEvent('click', { view: window, bubbles: true, cancelable: true });
```
Comment (artembelik) [inline packages/primitives/popover/src/popover-anchor.directive.ts:47]: you need to `inject(DOCUMENT)`:

```suggestion
        const clickEvent = new MouseEvent('click', { view: this.document.defaultView, bubbles: true, cancelable: true });
```

File: packages/primitives/popover/src/popover-content.directive.ts [hunk 2]
```diff
@@ -153,15 +156,34 @@ export class RdxPopoverContentDirective implements OnInit {
...
+                filter((event) => {
+                    return (
+                        !this.popoverRoot.popoverAnchorDirective() ||
+                        !this.popoverRoot
+                            .popoverTriggerDirective()
+                            .elementRef.nativeElement.contains(event.target as Element)
+                    );
+                }),
+                tap((event) =>
+                    console.log(
```
Comment (artembelik) [inline packages/primitives/popover/src/popover-content.directive.ts:168]: cleaning reminder :)

File: packages/primitives/popover/src/popover-anchor.directive.ts (hunk 1 above)
Comment (pawel-twardziak) [reply packages/primitives/popover/src/popover-anchor.directive.ts:47]: yep, sure thing :)

[the remark this answers is not in the document]
Comment (pawel-twardziak) [reply packages/primitives/popover/src/popover-anchor.directive.ts:47]: SRR will be groomed soonish :) there -> https://github.com/radix-ng/primitives/issues/202

File: packages/primitives/popover/src/popover-content.directive.ts (hunk 2 above)
Comment (pawel-twardziak) [reply packages/primitives/popover/src/popover-content.directive.ts:168]: yep, thank you :)

**Comment by pawel-twardziak** [comment]
@pimenovoleg  @artembelik ready to CR

**Comment by pimenovoleg** [comment]
@pawel-twardziak 

In the first example, when clicking on the trigger, the arrow is slightly shifted to the left. If I scroll down, the arrow becomes centered.

image

**Comment by pimenovoleg** [comment]
The message logs are duplicated between examples. Is it possible to isolate them?

image

image

**Comment by pawel-twardziak** [comment]
> @pawel-twardziak
> 
> In the first example (any example), when clicking on the trigger, the arrow is slightly shifted to the left. If I scroll down, the arrow becomes centered.
> image

thanks, will fix it @pimenovoleg

**Comment by pawel-twardziak** [comment]
> The message logs are duplicated between examples. Is it possible to isolate them? More precisely, they are not duplicated; instead, logs from Example A are being recorded in Example B.
> image image

@pimenovoleg you mean the outside click log? If so, that one comes from outside and all the stories catch it.

**Comment by pawel-twardziak** [comment]
@pimenovoleg regarding the logs. I've checked on them. It works as expected. The outside click event is catched by each popover if only the popover is open. If it is closed the outside click does not affect the popover.

**Comment by pawel-twardziak** [comment]
> @pawel-twardziak
> 
> In the first example (any example), when clicking on the trigger, the arrow is slightly shifted to the left. If I scroll down, the arrow becomes centered.
>

@pimenovoleg  I cannot reproduce it. What browser do you use? It migth be reated to the logs which are rendered at the same time the popover is being opened

**Comment by pimenovoleg** [comment]
> > @pawel-twardziak
> > In the first example (any example), when clicking on the trigger, the arrow is slightly shifted to the left. If I scroll down, the arrow becomes centered.
> 
> @pimenovoleg I cannot reproduce it. What browser do you use? It migth be reated to the logs which are rendered at the same time the popover is being opened

Chrome – Version 131.0.6778.70 (Official Build) (arm64)

I think this happens due to the appearance of a `scrollbar`, which slightly reduces the width and shifts the arrow. Perhaps it's not worth paying too much attention to this for now.
Shifted left on "Canvas" example, 
https://primitives-git-fork-pawel-twardziak-featur-81ec31-radix-angular.vercel.app/?_vercel_share=5eEwZkT6jg2a7057R9tFrn9qGgT4IG3A&path=%2Fdocs%2Fprimitives-popover--docs

But its ok on single example:
https://primitives-git-fork-pawel-twardziak-featur-81ec31-radix-angular.vercel.app/?path=%2Fstory%2Fprimitives-popover--default

**Comment by pawel-twardziak** [comment]
> > > @pawel-twardziak
> > > In the first example (any example), when clicking on the trigger, the arrow is slightly shifted to the left. If I scroll down, the arrow becomes centered.
> > 
> > 
> > @pimenovoleg I cannot reproduce it. What browser do you use? It migth be reated to the logs which are rendered at the same time the popover is being opened
> 
> Chrome – Version 131.0.6778.70 (Official Build) (arm64)
> 
> I think this happens due to the appearance of a `scrollbar`, which slightly reduces the width and shifts the arrow. Perhaps it's not worth paying too much attention to this for now. Shifted left on "Canvas" example, https://primitives-git-fork-pawel-twardziak-featur-81ec31-radix-angular.vercel.app/?_vercel_share=5eEwZkT6jg2a7057R9tFrn9qGgT4IG3A&path=%2Fdocs%2Fprimitives-popover--docs
> 
> But its ok on single example: https://primitives-git-fork-pawel-twardziak-featur-81ec31-radix-angular.vercel.app/?path=%2Fstory%2Fprimitives-popover--default

yes @pimenovoleg , the scrollbar can causes the issue. Ok, I will see if I can fix it in no time uncluding the scrollbar dimensions explicit. I not, will skip it for now.

**Pull request merged**